src/MICore/Transports/PipeTransport.cs

```diff
@@ -93,9 +93,11 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
         {
             PipeLaunchOptions pipeOptions = (PipeLaunchOptions)options;
 
+            string workingDirectory = pipeOptions.PipeCwd;
             if (!LocalLaunchOptions.CheckDirectoryPath(pipeOptions.PipeCwd))
```

Comment:
The one problem with this is that if the user provides a 'PipeCmd' attribute which is bogus, we will no longer throw. We might want to instead remove LaunchOptions.cs(82) and change this line to be:

```csharp
    if (!string.IsNullOrEmpty(workingDirectory))
    {
       if (!CheckDirectoryPath()..) throw;
    }
   else
   {
      workingDirectory = Path.GetDirectoryName(pipePath);
      if (!CheckDirectoryPath()..)
      {
        your new code
      }
```